### PR TITLE
DEV: use data type

### PR DESCRIPTION
**Pull request by licode**
Correction based on https://github.com/NSLS-II/suitcase/pull/21

### CODE REVIEW

File: suitcase/hdf5.py [hunk 1]
```diff
@@ -96,25 +96,23 @@ def export(headers, filename, stream_name=None, fields=None, timestamps=True, us
                                                 fletcher32=True)
                     data = [e['data'][key] for e in events]
                     data = np.array(data)
-                    try:
+                    if data.dtype.kind in ['S', 'U']:
```
Comment (tacaswell) [inline suitcase/hdf5.py]: This is still looking at the type of the array, we can look at the type reported by our descriptor.

Comment (licode) [inline suitcase/hdf5.py]: I tried to use the information in descriptors. It looks like 2D array of string and 2D array of number are both defined as 'array' in dtype. So I still need to use extra information like np.dtype to get the correct data type.

Comment (klauer) [inline suitcase/hdf5.py]: @licode what's the name of the key for the 2D string data? I'm curious where this is coming from (probably HXN...).

Comment (licode) [inline suitcase/hdf5.py]: Yes, it is from hxn. 

Please try runid=13123 and data_key='merlin1_stats1_asyn_pipeline_config'

Comment (tacaswell) [inline suitcase/hdf5.py]: That should not be going into read_attrs, but should be in the configuration instead.

Comment (danielballan) [inline suitcase/hdf5.py]: Fair enough, @tacaswell, but how can we move forward with this PR? I would suggest @licode check the self-reported dtype and then, if an error is caught, try inferring the actual datatype instead. Thoughts?

Comment (licode) [inline suitcase/hdf5.py]: Sounds good to me. I will modify this.

**Comment by licode** [comment]
Change on mds is updated for suitcase. more comments?

**Comment by danielballan** [comment]
I see that you added a metadatastore argument to the insert functions. This is a good change. You made it optional with a default value of `db` from `from databroker import db`. Unfortunately, if there are no global configuration files, that import will fail. If we want these insert functions to work going forward, I think the only safe thing to do is make the metadatastore argument *required* with no default.

Also, as a minor point, the name "mdsc" comes from "metadatastore.commands" which is now gone. Might as well name your new parameter `mds`, the same as we have named it elsewhere.

**Comment by licode** [comment]
Thanks for the suggestions. Also doc needs to be improved.

**Comment by licode** [comment]
remove default on db.mds. done.

**Comment by licode** [comment]
I think this is ready to go.

**Comment by licode** [comment]
more comments on this? @danielballan

File: suitcase/hdf5.py [hunk 2]
```diff
@@ -96,25 +96,27 @@ def export(headers, filename, stream_name=None, fields=None, timestamps=True, us
...
+                            for v in data[0]:
+                                data_len = max(data_len, len(v))
+                            data = data.astype('|S'+str(data_len))
                             dataset = data_group.create_dataset(
                                 key, data=data, compression='gzip')
-                        except TypeError:
-                            # ignore other types
-                            print('Dataset {} is not saved, as the data type is {}'.format(key, data.dtype))
+                        else:
+                            raise('Array of str with ndim >= 3 can not be saved.')
```
Comment (danielballan) [inline suitcase/hdf5.py]: Python doesn't let you `raise` without somehow providing a specific type of exception. See for example:

```python
In [1]: raise('asdf')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-be09e45a66d7> in <module>()
----> 1 raise('asdf')

TypeError: exceptions must derive from BaseException
```

Instead do something like:

```python
In [2]: raise ValueError("asdf")
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-ccfb515dbf45> in <module>()
----> 1 raise ValueError("asdf")

ValueError: asdf
```

**Comment by danielballan** [comment]
Sorry for the delay, @licode. Fix the one syntax problem and then self-merge this at will.

**Comment by licode** [comment]
good catch. thanks!

**Pull request merged**